### PR TITLE
Add ability to disable panning in the OrbitManipulator

### PR DESCRIPTION
**Pull request by mentholyspirit**
It's useful when you want to have the ability to rotate the camera, but want to limit panning.

### CODE REVIEW

File: libs/camutils/include/camutils/Manipulator.h [hunk 1]
```diff
@@ -143,6 +144,8 @@ class CAMUTILS_PUBLIC Manipulator {
         Builder& groundPlane(FLOAT a, FLOAT b, FLOAT c, FLOAT d);  //! Plane equation used as a raycast fallback
         Builder& raycastCallback(RayCallback cb, void* userdata);  //! Raycast function for accurate grab-and-pan
 
+        Builder& panDisabled(bool enabled);  //! Sets whether panning with two fingers is enabled
```
Comment (poweifeng) [inline libs/camutils/include/camutils/Manipulator.h:147]: I'd remove the "two fingers" part, which only applies to mobile.

**Review by poweifeng — APPROVED** [review]

File: libs/camutils/include/camutils/Manipulator.h [hunk 2]
```diff
@@ -143,6 +144,8 @@ class CAMUTILS_PUBLIC Manipulator {
         Builder& groundPlane(FLOAT a, FLOAT b, FLOAT c, FLOAT d);  //! Plane equation used as a raycast fallback
         Builder& raycastCallback(RayCallback cb, void* userdata);  //! Raycast function for accurate grab-and-pan
 
+        Builder& panDisabled(bool enabled);  //! Sets whether panning is enabled
```
Comment (pixelflinger) [inline libs/camutils/include/camutils/Manipulator.h:147]: rename to `panning`

File: android/filament-utils-android/src/main/java/com/google/android/filament/utils/Manipulator.java [hunk 3]
```diff
@@ -274,6 +274,17 @@ public Builder groundPlane(float a, float b, float c, float d) {
             return this;
         }
 
+        /**
+         * Sets whether panning is enabled in the manipulator.
+         *
+         * @return this <code>Builder</code> object for chaining calls
+         */
+        @NonNull
+        public Builder panDisabled(Boolean enabled) {
```
Comment (pixelflinger) [inline android/filament-utils-android/src/main/java/com/google/android/filament/utils/Manipulator.java:283]: rename to `panning`

File: libs/camutils/include/camutils/Manipulator.h [hunk 4]
```diff
@@ -109,6 +109,7 @@ class CAMUTILS_PUBLIC Manipulator {
         vec4 groundPlane;
         RayCallback raycastCallback;
         void* raycastUserdata;
+        bool panDisabled;
```
Comment (pixelflinger) [inline libs/camutils/include/camutils/Manipulator.h:112]: rename to `panning`

File: libs/camutils/src/Manipulator.cpp [hunk 5]
```diff
@@ -144,6 +144,14 @@ Manipulator<FLOAT>::Builder& Manipulator<FLOAT>::Builder::raycastCallback(RayCal
     return *this;
 }
 
+
+
+template <typename FLOAT> typename
+Manipulator<FLOAT>::Builder& Manipulator<FLOAT>::Builder::panDisabled(bool enabled) {
+    details.panDisabled = enabled;
```
Comment (pixelflinger) [inline libs/camutils/src/Manipulator.cpp:151]: this is *very* confusing. It should all be done in terms of "panning being enabled or not".

**Review by pixelflinger — CHANGES_REQUESTED** [review]
rename `panDisabled`  to `panning(bool) where `true` means panning enabled and `false` means panning disabled. Make the default be `true`.

**Comment by mentholyspirit** [comment]
@pixelflinger The problem then comes when someone tries to use it like this:
`Config config = {};
config.upVector = vec2(1.f, 0.f);
mainpulator.setProperties(config)`
Then `panning` will be set to false, which is not the expected behaviour.

**Comment by pixelflinger** [comment]
> @pixelflinger The problem then comes when someone tries to use it like this: `Config config = {}; config.upVector = vec2(1.f, 0.f); mainpulator.setProperties(config)` Then `panning` will be set to false, which is not the expected behaviour.

Why can't we set `panning` to `true` by default in `Config`?

**Review by pixelflinger — APPROVED** [review]

**Comment by pixelflinger** [comment]
@mentholyspirit the Android presubmit is failing.

**Pull request merged**